app/src/ui/lib/filter-list.tsx

```diff
@@ -117,6 +117,9 @@ interface IFilterListProps<T extends IFilterListItem> {
 
   /** Called when there are no items to render.  */
   readonly renderNoItems?: () => JSX.Element | null
+
+  /** Used to get a reference to a TextBox that will be used to control this component */
+  readonly filterTextBox?: TextBox //Todo: don't rely on passing textbox to component
```

Comment:
Maybe I'm opening a can of worms here but I wonder if we can avoid a lot of these "if" statements and the extra component for `FancyTextBox` by adding an `icon` prop to the `TextBox` that can be null or set to something else (ex: the octicon). That way we don't have to pass in an override or deal with finding a creative way to `selectAll()` on either a `TextBox` or a `FancyTextBox`. 

I have a feeling I'm missing some nuance here, though.

Reply:
That was the original idea, but I felt it was not good to overload the `TextBox` component to do more than it does now. I'm have no strong opinions either way. One thing to keep in mind is that I created #4317 so that this can be refactored.

Reply:
👍